04-materials/05-developing-with-ai.md

```diff
@@ -50,6 +50,31 @@ You would never hire a developer and expect absolute perfection in their creativ
 - Treat errors as learning opportunities for both you and the AI
 - Don't be afraid to roll back to the beginning and start over if AI doubled down on a wrong path
 
+### ✍️ Prompt fundamentals (Cursor)
+
+Keep prompts short, specific, and grounded in the exact code you're changing.
+
+- **Intent + state**: State what you want and show the current code (select lines or attach files)
+- **Point with `@`**: Reference files/folders/docs (e.g., `@src/index.ts`, `@plans/...`, `@JupyterLab docs`)
+- **Constrain**: Provide acceptance criteria and constraints (APIs to use, error handling)
+- **Iterate**: Prefer small steps; ask follow‑ups over one giant prompt
+- **Selections win**: Selecting code before asking gives that text highest priority in context
+
+
+
+:::{dropdown} Examples (before → better)
+**Before:** "Add image editing."
+
+**Better:**
```

Comment:
This before and after improvement example is really neat!

Reply:
Awesome! Thanks 🎆